resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -39,57 +40,107 @@ function Get-ValidTimeZone {
+
+    if ($LASTEXITCODE -ne 0) {
+        throw [System.Configuration.ConfigurationException]::new("Failed to set daylight saving time. Error: $LASTEXITCODE")
+    }
+}
 #endRegion Functions
 
 #region Classes
 <#
 .SYNOPSIS
-    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+    This `TimeZone` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
 
 .DESCRIPTION
-    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+    This `TimeZone` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
 
-.PARAMETER TimeZone
-    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+.PARAMETER Id
+    The Id to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
 
 .PARAMETER SetTimeZoneAutomatically
-    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
+    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time zone automatically.
+
+.PARAMETER SetTimeAutomatically
+    Whether to set the time automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
 
 .PARAMETER AdjustForDaylightSaving
     Whether to adjust for daylight saving time. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Adjust for daylight saving time automatically.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
 
     This example sets the time zone to Pacific Standard Time.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Get -Property {}
 
     This example gets the current time settings on the machine.
 #>
 [DscResource()]
 class TimeZone {
     [DscProperty(Key)]
-    [string] $TimeZone
+    [string] $Id
 
     [DscProperty()]
     [nullable[bool]] $SetTimeZoneAutomatically
 
+    [DscProperty()]
+    [nullable[bool]] $SetTimeAutomatically
+
```

Comment:
How do you feel about moving things like `SetTimeAutomatically` to a separate DSC resource called `Time`.
When I look at the settings page for TimeZone, I only see `SetTimeZoneAutomatically` and if that is not true, then we have the option to set the TimeZoneId and DaylightSaving. I think this DSC resource should try to replicate what we see in the settings app. I think this is the best way to prevent scope creep for this individual DSC resources.

Reply:
That's precisely why I did it into _one_ DSC resource. They bite each other. Even if we separate them both, we have to check what options are already enabled in both cases. If we set the Timezone to automatically, we _might_ need to clean up the manual settings and vice versa.